staging/src/k8s.io/code-generator/kube_codegen.sh

```diff
@@ -0,0 +1,585 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# shellcheck disable=2046 # printf word-splitting is intentional
```

Comment:
> We know the inputs do not contain spaces because we control them, but in general you raise a fair point.

This feels like the kind of assumption that breaks down later though.